projects/rocprofiler-systems/source/lib/core/trace_cache/rocpd_post_processing.cpp

```diff
@@ -375,6 +376,219 @@ rocpd_post_processing::get_pmc_event_with_sample_callback() const
     };
 }
 
+postprocessing_callback
+rocpd_post_processing::get_amd_smi_sample_callback() const
+{
+    struct xcp_metrics_t
+    {
+        std::vector<uint16_t> vcn_busy;
+        std::vector<uint16_t> jpeg_busy;
+    };
+
+    auto deserialize_xcp_metrics = [](const std::vector<uint8_t>& data,
+                                      xcp_metrics_t&              result) {
+        constexpr size_t elem_size = sizeof(uint16_t) / sizeof(uint8_t);
+
+        size_t  offset    = 0;
+        uint8_t vcn_size  = 0;
+        uint8_t jpeg_size = 0;
+
+        vcn_size = *(data.data() + offset);
+        offset++;
+        offset += vcn_size * elem_size;
+
+        jpeg_size = *(data.data() + offset);
+
+        if(data.size() != (2 + ((vcn_size + jpeg_size) * elem_size)))
+        {
+            throw std::invalid_argument("XCP metric data is corrupted");
+        }
+
+        result.vcn_busy.resize(vcn_size);
+        result.jpeg_busy.resize(jpeg_size);
+
+        const auto* vcn_data  = data.data() + 1;
+        const auto* jpeg_data = data.data() + 2 + vcn_size;
```

Comment:
```suggestion
        const auto* jpeg_data = data.data() + 2 + (vcn_size * elem_size);
```
Shouldn't it be multiplied by the element size?